Add support for Azure CLI authentication

sebastienros opened the pull request.

sebastienros commented:
Only service bus relay for now. SQL Azure will follow.

sebastienros commented on src/Microsoft.Crank.Controller/Program.cs:
```diff
@@ -3372,35 +3372,40 @@ private static ExecutionResult ComputeAverages(IEnumerable<ExecutionResult> exec
             return executionResult;
         }
 
-        static TokenProvider GetAadTokenProvider(string clientId, string tenantId, X509Certificate2 cert)
```
We do use cert based auth for SQL Server, but this method was no used.

DrewScoggins commented:
I am curious what the plan is for this long term? Do we imagine we would have some kind of security group that we could add people to for the proper access to the relay?

Question aside the code LGTM.

sebastienros commented:
I added that to be able to use Managed Identities. In ADO there is a service connection for a user managed identity, and with the AzureCliTask we can use this service connection. So in theory we could use the same principle for SQL Server when running on ado. Then the fact it works locally helps with debugging. For instance I added myself as a sender on the relay and it worked. If we want to create a group we can too, but I don't see a reason for now (unless we want to send jobs on the Azure VMs directly since they are in a private vlan).
